contracts/token/Faucet.sol

```diff
@@ -105,6 +125,76 @@ contract Faucet {
         emit FaucetDripped(_recipient);
     }
 
+    /**
+     * @notice Drip ERC20 tokens to a list of addresses as a batch to decrease gas costs.
+     * There is no check to see if the addresses have already been dripped to. Use with caution.
+     * @dev This function is a modified version of the airdropERC20 function from GasliteDrop.sol https://etherscan.io/address/0x09350f89e2d7b6e96ba730783c2d76137b045fef#code
+     *
+     * @param _token The address of the ERC20 contract
+     * @param _addresses The addresses to airdrop to
+     * @param _amounts The amounts to airdrop
+     * @param _totalAmount The total amount to airdrop
+     */
+    function batchDrip(
+        address _token,
+        address[] calldata _addresses,
+        uint256[] calldata _amounts,
+        uint256 _totalAmount
+    ) external payable isApprovedOperator {
+        emit BatchDrip(_addresses.length, _totalAmount);
+
+        assembly {
+            // Check that the number of addresses matches the number of amounts
+            if iszero(eq(_amounts.length, _addresses.length)) {
```

Comment:
this isn't significantly more efficient to do in assembly

Reply:
moved it